coins/monero/tests/wallet-rpc-compatibility.rs

```diff
@@ -0,0 +1,88 @@
+use std::{
+  collections::{HashSet, HashMap},
+  str::FromStr,
+};
+
+use rand_core::OsRng;
+use rand::RngCore;
+
+use monero_rpc::{
+  monero::{Amount, Address},
+  TransferOptions,
+};
+
+use monero_serai::{
+  wallet::address::{Network, AddressSpec, SubaddressIndex},
+  wallet::Scanner,
+};
+
+mod runner;
+
+async fn test_from_wallet_rpc_to(spec: AddressSpec) {
+  let wallet_rpc =
+    monero_rpc::RpcClientBuilder::new().build("http://127.0.0.1:6061").unwrap().wallet();
+  let daemon_rpc = runner::rpc().await;
+
+  // initialize wallet rpc
+  let address_resp = wallet_rpc.get_address(0, None).await;
+  let wallet_rpc_addr = if address_resp.is_ok() {
+    address_resp.unwrap().address
+  } else {
+    wallet_rpc.create_wallet("test_wallet".to_string(), None, "English".to_string()).await.unwrap();
```

Comment:
Should we always create a fresh wallet? Either is fine in this case. Just a commentary on providing fresh contexts/reusable code.

Reply:
Thought about it but always creating a fresh wallet per test will need extra boilerplate code to manage the wallet-rpc. Things like switch wallet, refresh etc.. Also will need to mine 70 blocks per tx instead of 10. Either is fine as you said thought going with single one is simpler/faster.